vital/algo/metadata_filter.cxx

```diff
@@ -14,12 +14,33 @@ namespace vital {
 
 namespace algo {
 
+const algorithm_capabilities::capability_name_t
+metadata_filter::CAN_USE_FRAME_IMAGE( "can-use-frame-image" );
+
+// ----------------------------------------------------------------------------
 metadata_filter
 ::metadata_filter()
 {
   attach_logger( "algo.metadata_filter" ); // specify a logger
 }
 
+// ----------------------------------------------------------------------------
+algorithm_capabilities const&
+metadata_filter
+::get_implementation_capabilities() const
+{
+  return m_capabilities;
+}
+
+// ----------------------------------------------------------------------------
+void
```

Comment:
Not super relevant, but I'm curious why this needs to be recreated here? It seems very much like boilerplate.

Reply:
I'm inclined to agree. Unfortunately, the capabilities API is not part of the base class, so this gets duplicated in every algorithm that supports capabilities. (Maybe we should refactor this? But maybe not in this PR...)

Reply:
Yeah, I'd agree this PR is not the place for it.